### PR TITLE
Fix code splitting

### PR DESCRIPTION
**Pull request by sandervspl**

### CODE REVIEW

**Review by stefanweck — APPROVED** [review]

File: config/webpack.config.base.ts [hunk 1]
```diff
@@ -66,6 +70,13 @@ const baseConfig: webpack.Configuration = {
           },
         ],
       },
+      // Fixes a @babel/runtime resolve issue https://github.com/webpack/webpack/issues/11467#issuecomment-691873586
+      {
+        test: /\.m?js/,
```
Comment (rnnyrk) [inline config/webpack.config.base.ts:75]: Shouldn't `.mjs` be also added as an extension resolver?

File: tsconfig.json [hunk 2]
```diff
@@ -8,7 +8,9 @@
     "skipLibCheck": true,
     "resolveJsonModule": true,
     "strict": true,
-    "module": "commonjs",
+    "removeComments": false,
```
Comment (rnnyrk) [inline tsconfig.json:11]: Why would you not want to remove comments?

Comment (rnnyrk) [reply tsconfig.json:11]: Or is this to keep the Webpack code splitting comments within import statements?

File: config/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "es5",
+    "esModuleInterop": true,
+    "resolveJsonModule": true
+  }
+}
```
Comment (rnnyrk) [inline config/tsconfig.json:8]: I found this article https://skovy.dev/code-splitting-with-webpack-typescript-react/ on having a separate `tsconfig.json` when writing Webpack configs in Typescript. Although it feels dirty, it makes sense. What you think about splitting the common chunks as well, as mentioned in "Optimizing Split Chunks" in that article?

File: tsconfig.json (hunk 2 above)
Comment (sandervspl) [reply tsconfig.json:11]: Webpack adds comments for code-splitting to work. So we need to preserve them in the transpiled code

File: config/tsconfig.json (hunk 3 above)
Comment (sandervspl) [reply config/tsconfig.json:8]: That's the article I used as a reference actually! Common chunks is something I could look at in the future, for sure.

File: config/webpack.config.base.ts (hunk 1 above)
Comment (sandervspl) [reply config/webpack.config.base.ts:75]: Tested dev and build and it worked with only this. Doesn't appear to be necessary to be added to extension resolving.

**Review by rnnyrk — APPROVED** [review]

**Review by JBostelaar — APPROVED** [review]

**Pull request merged**